### PR TITLE
Add CJS to build

### LINKED ISSUE #2: Cannot run package
Hi @bodil ,

Super cool thing you've made! Works amazing and removes all of the hacky result types I've been using myself

Your current published package.json looks like this:

<details><summary>Current</summary>
<pre>
<code>
{
  "module": "./esm/mod.js",
  "types": "./types/mod.d.ts",
  "name": "@bodil/opt",
  "version": "0.2.2",
  "description": "Option types for TypeScript with real gradual typing.",
  "license": "MPL-2.0+",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bodil/opt.git"
  },
  "bugs": {
    "url": "https://github.com/bodil/opt/issues"
  },
  "exports": {
    ".": {
      "import": {
        "types": "./types/mod.d.ts",
        "default": "./esm/mod.js"
      }
    }
  },
  "scripts": {
    "test": "node test_runner.js"
  },
  "devDependencies": {
    "@types/node": "^18.11.9",
    "chalk": "^4.1.2",
    "@deno/shim-deno": "~0.15.0"
  }
}
</code>
</pre>
</details> 

But that gives the following error when working with node v20.10.0

```
node:internal/modules/esm/resolve:294
  return new ERR_PACKAGE_PATH_NOT_EXPORTED(
         ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /myPath/node_modules/@bodil/opt/package.json
    at exportsNotFound (node:internal/modules/esm/resolve:294:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:584:13)
    at resolveExports (node:internal/modules/cjs/loader:591:36)
    at Module._findPath (node:internal/modules/cjs/loader:668:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1130:27)
    at resolve (/myPath/node_modules/tsx/dist/register-DfubRCxM.cjs:1:3084)
    at resolveRequest (/myPath/node_modules/tsx/dist/register-DfubRCxM.cjs:1:2618)
    at /myPath/node_modules/tsx/dist/register-DfubRCxM.cjs:1:3400
    at m._resolveFilename (file:///myPath/node_modules/tsx/dist/register-CFO5XQXL.mjs:1:832)
    at Module._load (node:internal/modules/cjs/loader:985:27) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

Somehow updating the package.json to this:

<details><summary>Working version</summary>
<pre>
<code>
{
  "module": "./esm/mod.js",
  "types": "./types/mod.d.ts",
  "name": "@bodil/opt",
  "version": "0.2.2",
  "description": "Option types for TypeScript with real gradual typing.",
  "license": "MPL-2.0+",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bodil/opt.git"
  },
  "bugs": {
    "url": "https://github.com/bodil/opt/issues"
  },
  "exports": {
    ".": {
      "import": "./esm/mod.js",
      "types": "./types/mod.d.ts",
      "require": "./esm/mod.js"
    }
  },
  "scripts": {
    "test": "node test_runner.js"
  },
  "devDependencies": {
    "@types/node": "^18.11.9",
    "chalk": "^4.1.2",
    "@deno/shim-deno": "~0.15.0"
  }
}
</code>
</pre>
</details> 

Fixes the error.

I looked at other existing package.json files and [this](https://stackoverflow.com/questions/62246824/error-err-package-path-not-exported-no-exports-main-resolved-in-app-node-m) StackOverflow post

### PR DESCRIPTION
**Pull request by mjwsteenbergen**
Closes #2

### CODE REVIEW

File: scripts/build_npm.ts [hunk 1]
```diff
@@ -25,14 +25,13 @@ if (
...
 await build({
-    scriptModule: false,
     entryPoints: ["./mod.ts"],
     outDir: "./npm",
     shims: {
         // see JS docs for overview and more options
         deno: "dev",
     },
-    compilerOptions: { lib: ["esnext", "dom"] },
+    compilerOptions: { lib: ["ESNext", "DOM"] },
```
Comment (mjwsteenbergen) [inline scripts/build_npm.ts:35]: The IDE was complaining about this

File: scripts/build_npm.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // ex. scripts/build_npm.ts
-import { build, emptyDir } from "https://deno.land/x/dnt@0.35.0/mod.ts";
+import { build, emptyDir } from "https://deno.land/x/dnt@0.40.0/mod.ts";
```
Comment (mjwsteenbergen) [inline scripts/build_npm.ts:2]: Just upgraded this for fun

File: scripts/build_npm.ts [hunk 3]
```diff
@@ -102,8 +101,9 @@ await new Deno.Command("npm", {
     stdout: "inherit",
     stderr: "inherit",
 }).output();
+
 await new Deno.Command("npm", {
-    args: ["exec", "typedoc"],
+    args: ["exec", "typedoc", "--", "--entryPointStrategy", "expand", "./src"],
```
Comment (mjwsteenbergen) [inline scripts/build_npm.ts]: I don't know if this gives you what you want, but at least it is no longer crashing

File: scripts/build_npm.ts [hunk 4]
```diff
@@ -25,14 +25,14 @@ if (
...
 await build({
-    scriptModule: false,
     entryPoints: ["./mod.ts"],
     outDir: "./npm",
     shims: {
         // see JS docs for overview and more options
         deno: "dev",
     },
-    compilerOptions: { lib: ["esnext", "dom"] },
+    declaration: "separate",
```
Comment (mjwsteenbergen) [inline scripts/build_npm.ts:34]: Without this line you no longer have types

**Comment by bodil** [comment]
Looks good, thanks, will merge and tag a release.

**Pull request merged**